dot/peerset/peerset.go

```diff
@@ -282,6 +291,7 @@ func (ps *PeerSet) updateTime() error {
 			after := reputationTick(before)
 			n.setReputation(after)
 			ps.peerState.nodes[peerID] = n
+			reputationGauge.WithLabelValues(peerID.Pretty()).Set(float64(after))
```

Comment:
That will potentially blow up  memory usage of the Prometheus server since peerID is rather random. Maybe leave this as a log or find another way to report the metric with a limited-space label.

Reply:
Ideally I only want to label this on the devnets where the peers are bound to only like 3-4 nodes.  I'll refactor this PR to make this configurable.

Reply:
Maybe just have it as a periodic log? 🤔 Like our sync benchmarker?

Or you could have a mapping from [1 to 50] to peerIDs, use the [1-50] set for the metrics labels and log out the mapping at program start. Not totally ideal either, but that would work.